lib/datadog/ci/configuration/components.rb

```diff
@@ -17,13 +17,6 @@ def initialize(settings)
         def activate_ci_mode!(settings)
           # Activate underlying tracing test mode
           settings.test_mode.enabled = true
-
-          # Choose user defined TraceFlush or default to CI TraceFlush
-          settings.test_mode.trace_flush = settings.ci_mode.trace_flush \
-                                             || Datadog::CI::TraceFlush::Finished.new
-
-          # Pass through any other options
-          settings.test_mode.writer_options = settings.ci_mode.writer_options
         end
       end
     end
```

Comment:
As usual, fully OK if you want to separate this out to another PR, but is it just me or the whole point of this `module` was to transform a `settings.ci_mode.enabled` into three other settings (all under `test_mode`).

Now that we mostly deleted that, is it just me or does this whole module become a really awkward way of setting `settings.test_mode.enabled` based on `settings.ci_mode.enabled`?

I think we should be able to simplify setting one from the other, but I wonder if we could even go one step further and remove (or at least deprecate) one of the two options, perhaps?

What's the difference between `settings.test_mode` and `settings.ci_mode` anyway? I can't seem to find any public references to `settings.test_mode` 😅

Reply:
I'm not 100% sure on the background here, but difference is that `c.ci_mode ...` activates the whole CI-App suite, which can be more than `c.test_mode.enabled = true`. Or at least it was before this change.

I think the goal is to line up `c.ci_mode ...` with the single Env var that activates CI-App (we really like a single Env var that enables products around here).

I'm not strongly tied to either `c.test_mode` or `c.ci_mode`, tbh.

@delner would you recall why the options are organized the way they are.
Or maybe @juan-fernandez: would these option names be standard across different languages?

Reply:
it's difficult to have a standard for every language. `ci_mode` or `ci_visibility_enabled` or something of the sort is fine